scripts/ir_to_markdown.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Converts a dump of IR text to a markdown file with formatting.
+
+Example usage:
+  # Get a dump of IR from a compiler tool:
+  $ iree-translate \
+    -iree-transformation-pipeline \
+    -iree-hal-target-backends=vmvx \
+    -mlir-disable-threading \
+    -print-ir-after-all \
+    -mlir-elide-elementsattrs-if-larger=8 \
+    $PWD/iree/samples/models/simple_abs.mlir \
+    -o /tmp/simple_abs_vmvx.vmfb \
+    2> /tmp/simple_abs_vmvx_pipeline.mlir
+
+  # Convert the IR dump to markdown:
+  $ python3 ir_to_markdown.py \
+    /tmp/fullyconnected_vmvx_pipeline.mlir \
+    -o /tmp/fullyconnected_vmvx_pipeline.md
+"""
+
+import argparse
+import re
+
+MLIR_START_SEQUENCE = "// -----//"
+MLIR_END_SEQUENCE = "//----- //"
+
+
+def parse_arguments():
+  """Parses command line arguments."""
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument('input_file_path',
+                      type=str,
+                      nargs='?',
+                      metavar="<input_file_path>",
+                      help='Input IR dump (.mlir from -print-ir-after-all)')
+  parser.add_argument('-o,',
+                      '--output',
+                      type=str,
+                      required=True,
+                      metavar="<output>",
+                      help='Output file path (e.g. translation_ir.md)')
+  # TODO(scotttodd): flag for input IR path
+  # TODO(scotttodd): flag for command line used for compilation
+  # TODO(scotttodd): flag for markdown flavor (mkdocs, github, etc.)
+  # TODO(scotttodd): flag for diff view (correlate IR before and IR after)?
```

Comment:
BTW, I prototyped this in a gist: https://gist.github.com/ScottTodd/191edcc56bbc7998ac0c2704c9793863

Outside of GitHub flavored markdown we have more options for how to display diffs.